dev/breeze/src/airflow_breeze/commands/release_candidate_command.py

```diff
@@ -190,16 +190,16 @@ def prepare_pypi_packages(version, version_suffix, repo_root):
         console_print("PyPI packages prepared")
 
 
-def push_packages_to_test_pypi():
+def push_packages_to_test_pypi(version):
     if confirm_action("Do you want to push packages to test PyPI?"):
         run_command(["twine", "upload", "-r", "pypitest", "dist/*"], dry_run_override=DRY_RUN, check=True)
         console_print("Packages pushed to test PyPI")
         console_print(
             "Verify that the test package looks good by downloading it and installing it into a virtual "
             "environment. The package download link is available at: "
-            "https://test.pypi.org/project/apache-airflow/#files"
-            "Install it with the appropriate constraint file, for instance: "
-            "pip install <download_link> --constraint https://raw.githubusercontent.com/apache/airflow/constraints-2.2.1rc1/constraints-3.8.txt"  # noqa: 501
+            "https://test.pypi.org/project/apache-airflow/#files "
+            "Install it with the appropriate constraint file, adapt python version: "
+            f"pip install https://test.pypi.org/simple/ apache-airflow=={version} --constraint https://raw.githubusercontent.com/apache/airflow/constraints-{version}/constraints-3.8.txt"  # noqa: 501
```

Comment:
Did this work for you? I got the below error:
```
(env) ➜  airflow pip install https://test.pypi.org/simple/ apache-airflow==2.5.2rc2 --constraint https://raw.githubusercontent.com/apache/airflow/constraints-2.5.2rc1/constraints-3.8.txt
Collecting https://test.pypi.org/simple/
  Using cached https://test.pypi.org/simple/ (9.1 MB)
  ERROR: Cannot unpack file /private/var/folders/tj/djvjnrjj5ll7f0vct7z6fb3h0000gp/T/pip-unpack-8gxtsyp7/simple.html (downloaded from /private/var/folders/tj/djvjnrjj5ll7f0vct7z6fb3h0000gp/T/pip-req-build-cifsze5i, content-type: text/html); cannot detect archive format
ERROR: Cannot determine archive format of /private/var/folders/tj/djvjnrjj5ll7f0vct7z6fb3h0000gp/T/pip-req-build-cifsze5i
```

Reply:
Can you try with `apache-airflow==2.5.2rc1` ? rc number seems wrong

Reply:
Indeed i'm missing the `-i` option that got lost in the process, updating

Reply:
Fixed, thanks :)

Reply:
I still got errors:
```
(env) ➜  airflow pip install -i https://test.pypi.org/simple/ apache-airflow==2.5.2rc1 --constraint https://raw.githubusercontent.com/apache/airflow/constraints-2.5.2rc1/constraints-3.9.txt
Looking in indexes: https://test.pypi.org/simple/
Collecting apache-airflow==2.5.2rc1
  Using cached https://test-files.pythonhosted.org/packages/d7/e6/2032cc1e9409f1b332ffe6831a9fb90d3d62c86a4d3add35b81baa4a341c/apache_airflow-2.5.2rc1-py3-none-any.whl (11.6 MB)
ERROR: Cannot install apache-airflow==2.5.2rc1 because these package versions have conflicting dependencies.

The conflict is caused by:
    apache-airflow 2.5.2rc1 depends on rfc3339-validator>=0.1.4
    The user requested (constraint) rfc3339-validator==0.1.4

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```

Reply:
You are likely runnint it with a different Python version.

Reply:
(you need to match the constraints python version with the local version you have)

Reply:
@ephraimbuddy

I can't reproduce with:
```
pip install -i https://test.pypi.org/simple/ apache-airflow==2.5.2rc1 --constraint https://raw.githubusercontent.com/apache/airflow/constraints-2.5.2rc1/constraints-3.9.txt
```

Can you check if your pip version is upgraded in your virtualenv ? (pip install --upgrade pip)

Reply:
I think I have the answer @ephraimbuddy.

I believe this had something to do with what @potiuk mentionned before. When installing only from `test.pypi.org` some packages might not be found because they do not exist on that test index. (I had trouble to reproduce because once those packages are cached by a successful install, they will be used and the index will not be used. Starting from a fresh env and purging cache highlights the issue).

For instance in this case [rfc3339-validator==0.1.4](https://test.pypi.org/project/rfc6920/#history) does not exist on test index.

I will update the PR to use the default index as a fallback (it doesn't work exactly as a fallback but that's the idea):
```
pip install -i https://test.pypi.org/simple/ --extra-index-url https://pypi.org/simple/ apache-airflow==2.5.2rc2 --constraint https://raw.githubusercontent.com/apache/airflow/constraints-2.5.2rc2/constraints-3.8.txt
```

Reply:
Just updated, let me know if this fixes the issues on your end :)